internal/ingress/controller/nginx_test.go

```diff
@@ -89,6 +95,72 @@ func TestIsDynamicConfigurationEnough(t *testing.T) {
 	}
 }
 
+func TestConfigureDynamically(t *testing.T) {
+	backends := []*ingress.Backend{{
+		Name: "fakenamespace-myapp-80",
+		Endpoints: []ingress.Endpoint{
+			{
+				Address: "10.0.0.1",
+				Port:    "8080",
+				Target:  &apiv1.ObjectReference{},
+			},
+			{
+				Address: "10.0.0.2",
+				Port:    "8080",
+				Target:  &apiv1.ObjectReference{},
+			},
+		},
+	}}
+
+	servers := []*ingress.Server{{
+		Hostname: "myapp.fake",
+		Locations: []*ingress.Location{
+			{
+				Path:    "/",
+				Backend: "fakenamespace-myapp-80",
+			},
+		},
+	}}
+
+	commonConfig := &ingress.Configuration{
+		Backends: backends,
+		Servers:  servers,
+	}
+
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusCreated)
+
+		if r.Method != "POST" {
+			t.Errorf("expected a 'POST' request, got '%s'", r.Method)
+		}
+
+		b, err := ioutil.ReadAll(r.Body)
+		if err != nil && err != io.EOF {
+			t.Fatal(err)
+		}
+		body := string(b)
+		if strings.Index(body, "target") != -1 {
+			t.Errorf("unexpected target reference in JSON content: %v", body)
+		}
+
+		if strings.Index(body, "service") != -1 {
+			t.Errorf("unexpected service reference in JSON content: %v", body)
+		}
+
+	}))
+	defer ts.Close()
+
+	t.Logf("%v", commonConfig.Backends[0].Endpoints[0])
+	err := configureDynamically(commonConfig, ts.URL)
+	if err != nil {
+		t.Errorf("unexpected error posting dynamic configuration: %v", err)
+	}
+	t.Logf("%v", commonConfig.Backends[0].Endpoints[0])
+	if commonConfig.Backends[0].Endpoints[0].Target == nil {
```

Comment:
What about comparing this with its expected value `&apiv1.ObjectReference{}`(from https://github.com/kubernetes/ingress-nginx/pull/2365/files#diff-b43320e0cdcbbb75b01313d88f637165R105) instead? IMHO that would be a complete check since it covers the case where target is being changed to something other than `nil`